app/scripts/modules/core/src/pipeline/config/stages/evaluateVariables/EvaluateVariablesStageConfig.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { map, set } from 'lodash';
+
+import { IStageConfigProps, StageConfigField, MapEditor } from 'core';
+
+export interface IEvaluateVariablesStageConfigState {
+  variables: any;
+}
+
+export class EvaluateVariablesStageConfig extends React.Component<
+  IStageConfigProps,
+  IEvaluateVariablesStageConfigState
+> {
+  private expand(variables: any) {
+    return map(variables, (value, key) => ({ key, value }));
+  }
+
+  private stageFieldChanged = (fieldIndex: string, value: any) => {
+    set(this.props.stage, fieldIndex, this.expand(value));
```

Comment:
is this mutation necessary because of angularjs?

Reply:
I don't have the context on why it's done this way and I'm not sure it's entirely necessary, but it feels very awkward. I was going to look at whether there's a better alternative next.